rpc-server/src/metrics.rs

```diff
@@ -60,608 +68,133 @@ impl OptimisticUpdating {
-    // QUERY.view_state
-    pub(crate) static ref QUERY_VIEW_STATE_ERROR_0: IntCounter = try_create_int_counter(
-        "query_view_state_error_0",
-        "Query.view_state error 0: ReadRPC success, NEAR RPC success"
-    ).unwrap();
-
-    pub(crate) static ref QUERY_VIEW_STATE_ERROR_1: IntCounter = try_create_int_counter(
-        "query_view_state_error_1",
-        "Query.view_state error 1: ReadRPC success, NEAR RPC error"
-    ).unwrap();
-
-    pub(crate) static ref QUERY_VIEW_STATE_ERROR_2: IntCounter = try_create_int_counter(
-        "query_view_state_error_2",
-        "Query.view_state error 2: ReadRPC error, NEAR RPC success"
-    ).unwrap();
-
-    pub(crate) static ref QUERY_VIEW_STATE_ERROR_3: IntCounter = try_create_int_counter(
-        "query_view_state_error_3",
-        "Query.view_state error 3: ReadRPC error, NEAR RPC error"
-    ).unwrap();
-
-    pub(crate) static ref QUERY_VIEW_STATE_ERROR_4: IntCounter = try_create_int_counter(
-        "query_view_state_error_4",
-        "Query.view_state error 4: Failed to compare. Network or parsing error"
-    ).unwrap();
-    // end QUERY.view_state
-}
-
-lazy_static! {
-    // QUERY.function_call
-    pub(crate) static ref QUERY_FUNCTION_CALL_ERROR_0: IntCounter = try_create_int_counter(
-        "query_function_call_error_0",
-        "Query.function_call error 0: ReadRPC success, NEAR RPC success"
-    ).unwrap();
-
-    pub(crate) static ref QUERY_FUNCTION_CALL_ERROR_1: IntCounter = try_create_int_counter(
-        "query_function_call_error_1",
-        "Query.function_call error 1: ReadRPC success, NEAR RPC error"
-    ).unwrap();
+    pub(crate) static ref OPTIMISTIC_UPDATING: OptimisticUpdating = OptimisticUpdating::new();
 
-    pub(crate) static ref QUERY_FUNCTION_CALL_ERROR_2: IntCounter = try_create_int_counter(
-        "query_function_call_error_2",
-        "Query.function_call error 2: ReadRPC error, NEAR RPC success"
+    pub(crate) static ref REQUESTS_COUNTER: IntCounterVec = register_int_counter_vec(
+        "requests_counter",
+        "Total number of requests",
+        &["request_type"] // This declares a label named `request_type`
     ).unwrap();
 
-    pub(crate) static ref QUERY_FUNCTION_CALL_ERROR_3: IntCounter = try_create_int_counter(
-        "query_function_call_error_3",
-        "Query.function_call error 3: ReadRPC error, NEAR RPC error"
-    ).unwrap();
+    // Error metrics
+    // 0: ReadRPC success, NEAR RPC success"
+    // 1: ReadRPC success, NEAR RPC error"
+    // 2: ReadRPC error, NEAR RPC success"
+    // 3: ReadRPC error, NEAR RPC error"
+    // 4: Failed to compare. Network or parsing error"
 
-    pub(crate) static ref QUERY_FUNCTION_CALL_ERROR_4: IntCounter = try_create_int_counter(
-        "query_function_call_error_4",
-        "Query.function_call error 4: Failed to compare. Network or parsing error"
+    // QUERY
+    pub(crate) static ref QUERY_VIEW_ACCOUNT_ERRORS: IntCounterVec = register_int_counter_vec(
```

Comment:
I think you should be able to define a metric with more than 1 label, thus you can have something like

```
REQUEST_ERRORS: IntCounterVec = register_int_counter_vec(
  "request_errors_with_code",
  "Total number of errors for method with code",
  &["method", "error_code"]
).unwrap()
```

**Warning!** This is a pseudo-code please consult with docs to make it correct

Reply:
The above applies to similar places